File: my-ts-react-app/src/app/models/DiplomaThesis.ts

import { ConsultationSchedule } from './ConsultationSchedule';
import { Mentor } from './Mentor';
import { Student } from './Student';
import { Title } from './Title';

export interface DiplomaThesis {
  id: number;
  dueDate: Date | null;
  submissionDate: Date | null;
  assessment: number | null;
  level: string;
  studentId: number | null;
  student: Student | null;
  mentorId: number;
  mentor: Mentor;
  titleId: number;
  title: Title;
  consultationSchedules: ConsultationSchedule[];
}

File: Models/Entities/DiplomaThesis.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace DiplomaThesisDigitalization.Models.Entities
{
    public class DiplomaThesis
    {
        public int Id { get; set; }
       
        public DateTime? DueDate { get; set; } = null;
        public DateTime? SubmissionDate { get; set; } = null;
        public byte? Assessment { get; set; }
        public string Level { get; set; }

        [ForeignKey("Student")]
        public int? StudentId { get; set; }
        
        [JsonIgnore]
        public Student Student { get; set; }

        [ForeignKey("Mentor")]
        public int MentorId { get; set; }
        [JsonIgnore]
        public Mentor Mentor { get; set; }
        
        public int TitleID { get; set; }
        public Title Title { get; set; }

        public ICollection<ConsultationSchedule> ConsultationSchedules { get; set; }
    }
}
